FreeRTOS/FreeRTOS
# Driver checksum support for ICMP ping reply

Description
-----------

When using `ipconfigREPLY_TO_INCOMING_PINGS`, it does not currently support letting the driver calculate the ICMP checksum, even when `ipconfigDRIVER_INCLUDED_TX_IP_CHECKSUM` is set to 1. It fills up the ICMP checksum field with data, which messes with the hardware checksum calculation (at least, it did for me whilst using an STM32F429).

Test Steps
-----------

1. Set `ipconfigREPLY_TO_INCOMING_PINGS` to 1
2. Use a hardware checksum calculation for ICMP packets

Related Issue
-----------
I found an issue that was already open, created by someone else (#87).


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
